src/components/Layout/Header/Header.tsx

```diff
@@ -93,6 +96,24 @@ export const Header = memo(() => {
   )
 
   const currentWalletId = useSelector(selectWalletId)
+  const walletAccountIds = useSelector(selectWalletAccountIds)
+
+  useEffect(() => {
+    const isMetaMaskMultichainWallet = wallet instanceof MetaMaskShapeShiftMultiChainHDWallet
+    if (!isMetaMaskMultichainWallet) return
+
+    if (isSnapInstalled === false) {
+      // We have just detected that the user doesn't have the snap installed currently
+      // We need to check whether or not the user had previous non-EVM AccountIds and clear those
+      if (currentWalletId && walletAccountIds.some(accountId => isUtxoAccountId(accountId)))
+        appDispatch(portfolio.actions.clearWalletMetadata(currentWalletId))
+    }
+    // We don't want to re-run this every time `isSnapInstalled` reference changes
+    // By the time isMetaMaskMultichainWallet evaluates to true (i.e `wallet` is defined),
+    // `isSnapInstalled` will already be a non-null value
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
this introduces a race condition (though unlikely as you pointed out). I dont think there is much cost in allowing it to rerun whenever `isSnapInstalled` changes, especially if we add an early exit `if (isSnapInstalled !== false) return`. This way we can rely on typical react style without risk of race condition eventuating if logic etc changes

Reply:
If we're fine with this running this every 3 seconds, sure, happy to have the full deps here.
Agree this is theoretically a race condition, but in practice, the race winners are deterministic!

Reply:
Or what about this ec068bc1f0da3e4e8ed1ad2a0d9fb1595fc0a532

With the added memoization, this will still run every 3 seconds, but `walletAccountIds.some(accountId => isUtxoAccountId(accountId))` itself will be memoized, so all that will happen is some simple primitive checks now

Gave it a local sanity check and confirmed we're still 🛍️

Reply:
perfect 🤝